src/devices/machine/cs8900a.cpp

```diff
@@ -0,0 +1,1522 @@
+// license:GPL-2.0+
+// copyright-holders:Rhett Aultman
+/*************************************************************************
+
+    CS8900A ethernet controller implementation
+
+    by Rhett Aultman <roadriverrail@gmail.com>
+    ported to MAME from VICE Project (https://sourceforge.net/p/vice-emu/)
+    VICE CS8900 code by Spiro Trikaliotis <Spiro.Trikaliotis@gmx.de>
+
+**************************************************************************/
+
+#include <cstring>
+
+#include "machine/cs8900a.h"
+
+DEFINE_DEVICE_TYPE(CS8900A,  cs8900a_device,  "CS8900A",  "CA8900A ETHERNET IC")
```

Comment:
Short names must not use uppsercase letters – this would fail a validity check.
The description should be something like `"CS8900A Crystal LAN 10Base-T Ethernet MAC"` (based on its marketing name).